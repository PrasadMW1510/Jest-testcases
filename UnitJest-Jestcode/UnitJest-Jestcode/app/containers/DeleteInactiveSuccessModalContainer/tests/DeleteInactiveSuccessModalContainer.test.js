import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

import { DeleteInactiveSuccessModalContainer } from '../index';

describe('<DeleteInactiveSuccessModalContainer />', () => {
  let wrapper = null;

  let mockData = null;
  let mockHideModal = null;
  let mockGetInactiveCohortMembersRequest = null;

  let mockEvent = null;

  beforeEach(() => {
    mockData = { searchOpts: 'mockSearchOpts' };
    mockHideModal = jest.fn();
    mockGetInactiveCohortMembersRequest = jest.fn();

    mockEvent = { preventDefault: jest.fn() };

    wrapper = shallow(
      <DeleteInactiveSuccessModalContainer
        data={mockData}
        hideModal={mockHideModal}
        getInactiveCohortMembersRequest={mockGetInactiveCohortMembersRequest}
      />
    );
  });

  it('Expect to render correctly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('handleYes', () => {
    const deleteInactiveSuccessModal = wrapper.find('DeleteInactiveSuccessModal');
    deleteInactiveSuccessModal.prop('onYes')(mockEvent);

    expect(mockEvent.preventDefault).toHaveBeenCalled();
    expect(mockHideModal).toHaveBeenCalledTimes(2);
    expect(mockGetInactiveCohortMembersRequest).toHaveBeenCalledWith(mockData.searchOpts);
  });
});
