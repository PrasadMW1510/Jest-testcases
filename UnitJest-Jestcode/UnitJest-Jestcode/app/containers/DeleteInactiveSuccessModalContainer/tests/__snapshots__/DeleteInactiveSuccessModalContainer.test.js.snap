// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteInactiveSuccessModalContainer /> Expect to render correctly 1`] = `
<DeleteInactiveSuccessModal
  data={
    Object {
      "searchOpts": "mockSearchOpts",
    }
  }
  isOpen={true}
  onYes={[Function]}
/>
`;
